### ISSUE TITLE
[java] InefficientEmptyStringCheck misses String.trim().isEmpty()

### ISSUE THREAD

**Issue by jgerken** [pmd/pmd#290]
**Rule Set:** [java/strings.html#InefficientEmptyStringCheck](https://pmd.github.io/pmd-5.5.4/pmd-java/rules/java/strings.html#InefficientEmptyStringCheck)

**Description:**

Since Java 1.6 the `String` class provides the `isEmpty()` method which returns `true` if the string has a length of 0 ([JavaDoc](http://docs.oracle.com/javase/8/docs/api/java/lang/String.html#isEmpty--)). The check for the inefficient empty string check should be extended such that `String.trim().isEmpty()` triggers a violation, too.

**Code Sample demonstrating the issue:**

```java
final String str = "   test.  ";
// the next line should cause a violation of the InefficientEmptyStringCheck but does not
if (str.trim().isEmpty()) {
    System.out.println("String is empty");
}
```

**Running PMD through:** *Maven and Eclipse*

**Comment by jsotuyod** [comment]
@jgerken thanks for the report! You are absolutely right on this one.

**Comment by Oowekyala** [comment]
@jsotuyod I've made a PR for a fix

**Comment by jsotuyod** [comment]
@Oowekyala amazing, I'll review ASAP.

**Comment by jsotuyod** [comment]
Closed via #303, will be fixed for the next release cycle of PMD
